packages/babel-preset-env/src/targets-parser.js

```diff
@@ -109,6 +110,15 @@ type ParsedResult = {
 };
 const getTargets = (targets: Object = {}, options: Object = {}): Targets => {
   const targetOpts: Targets = {};
+
+  // `esmodules` as a target indicates the specific set of browsers supporting ES Modules.
+  // These values OVERRIDE the `browsers` field.
+  if (targets.esmodules) {
+    const supportsESModules = browserModulesData["es6.module"];
+    targets.browsers = Object.keys(supportsESModules)
+      .map(browser => `${browser} ${supportsESModules[browser]}`)
+      .join(", ");
+  }
```

Comment:
The main worry I'd have with this approach is that, at least for now, all those browsers are very fresh.

But I guess I'll only have to worry about cutting old browsers with this feature in 3~5 years so I guess it's a bit premature to worry.